enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+type Platform struct {
+    //...
+    // subnets specifies existing subnets (by ID) where cluster
+    // resources will be created.  Leave unset to have the installer
+    // create subnets in a new VPC on your behalf.
+    //
+    // Deprecated: Use platform.aws.vpc.subnets
+    //
+    // +optional
+    Subnets []string `json:"subnets,omitempty"`
+
+    // vpc specifies the VPC configuration.
+    //
+    // +optional
+    VPC VPCSpec `json:"vpc,omitempty"`
+}
+
+// VPCSpec configures the VPC for the cluster.
+type VPCSpec struct {
+    // subnets defines the subnets in an existing VPC
+    // and can optionally override their intended roles for use by the installer.
+    // If no roles are overridden on any subnet, then the subnet roles
+    // are decided automatically. In this case, the VPC must not contain
+    // any subnets without the kubernetes.io/cluster/<cluster-id> tag.
+    //
+    // For overridden subnet role selection, each subnet must have at
+    // least one assigned role, and all roles (ClusterNode, IngressControllerLB,
+    // ControlPlaneExternalLB, ControlPlaneInternalLB) must be assigned to
+    // at least one subnet.
+    // 
+    // Leave this field unset to have the installer create subnets
+    // in a new VPC on your behalf. subnets must contain unique IDs
+    // and must not contain more than 10 subnets with the IngressController
+    // role.
+    // subnets is only available in TechPreview.
+    //
+    // +optional
+    // +listType=atomic
+    // +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x.id == y.id))`,message="subnets cannot contain duplicate IDs" 
+    // +kubebuilder:validation:XValidation:rule=`self.exists(x, x.roles.exists(r, r == 'ClusterNode'))`,message="subnets must contain at least 1 subnet with the ClusterNode role"
+    // +kubebuilder:validation:XValidation:rule=`self.filter(x, x.roles.exists(r, r == 'IngressControllerLB')).size() <= 10`,message="subnets must contain less than 10 subnets with the IngressControllerLB role"
+    Subnets []Subnet `json:"subnets,omitempty"`
+}
+
+type Subnet struct {
+    // id specifies the subnet ID of an existing subnet.
+    // The subnet id must start with "subnet-", consist only
+    // of alphanumeric characters, and must be exactly 24
+    // characters long.
+    //
+    // +required
+    ID AWSSubnetID `json:"id"`
+
+    // rolesOverride manually overrides the roles (aka functions)
+    // that the subnet will provide in the cluster. If no roles are
+    // specified on any subnet, then the subnet roles are decided
+    // automatically. Each role must be unique.
+    //
+    // +optional
+    // +listType=atomic
+    // +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="rolesOverride cannot contain duplicates"
+    RolesOverride []SubnetRole `json:"rolesOverride"`
```

Comment:
Are any of the roles mutually exclusive? Eg can I have a subnets that is both CPExternalLB and CPInternalLB?

Reply:
Installer do generate a CRD from install-config for certain use cases, so the CEL isn't totally worthless